packages/framework/tests/Unit/Component/Money/MoneyTest.php

```diff
@@ -0,0 +1,582 @@
+        yield ['1.00', '1.000', '1.00000'];
+    }
+
+    public function testDivideIsImmutable(): void
+    {
+        $money = Money::fromString('1');
+
+        $money->divide('2', 1);
+
+        $this->assertEquals('1', $money->toString());
+    }
+
+    /**
+     * @dataProvider divideProvider
+     * @param string $a
+     * @param string $b
+     * @param int $scale
+     * @param string $expectedResult
+     */
+    public function testDivide(string $a, string $b, int $scale, string $expectedResult): void
+    {
+        $moneyA = Money::fromString($a);
+
+        $result = $moneyA->divide($b, $scale);
+
+        $this->assertEquals($expectedResult, $result->toString());
+    }
+
+    /**
+     * @return \Iterator
+     */
+    public function divideProvider(): Iterator
+    {
+        yield ['1', '2', 0, '1'];
+        yield ['1', '2', 1, '0.5'];
+        yield ['1', '2', 2, '0.50'];
+        yield ['1', '3', 3, '0.333'];
+        yield ['2', '3', 3, '0.667'];
+        yield ['1000', '3', 3, '333.333'];
+        yield ['3.33', '3', 2, '1.11'];
+        yield ['0.1', '0.1', 0, '1'];
+    }
+
+    public function testModIsImmutable(): void
+    {
+        $money = Money::fromString('1');
+
+        $money->mod(Money::fromString('1'));
+
+        $this->assertEquals('1', $money->toString());
+    }
+
+    /**
+     * @dataProvider modProvider
+     * @param string $a
+     * @param string $b
+     * @param string $expectedResult
+     */
+    public function testMod(string $a, string $b, string $expectedResult): void
+    {
+        $moneyA = Money::fromString($a);
+        $moneyB = Money::fromString($b);
+
+        $result = $moneyA->mod($moneyB);
+
+        $this->assertEquals($expectedResult, $result->toString());
+    }
+
+    /**
+     * @return \Iterator
+     */
+    public function modProvider(): Iterator
+    {
+        yield ['100', '100', '0'];
+        yield ['120', '100', '20'];
+        yield ['530', '100', '30'];
+        yield ['-20', '100', '80'];
+        yield ['-530', '100', '70'];
+        yield ['0.001', '100', '0.001'];
+        yield ['500.001', '100', '0.001'];
+        yield ['0.75', '0.5', '0.25'];
+    }
+
+    public function testRoundIsImmutable(): void
+    {
+        $money = Money::fromString('1.5');
+
+        $money->round(0);
+
+        $this->assertEquals('1.5', $money->toString());
+    }
+
+    /**
+     * @dataProvider roundProvider
+     * @param string $amount
+     * @param int $scale
+     * @param string $expectedResult
+     */
+    public function testRound(string $amount, int $scale, string $expectedResult): void
+    {
+        $money = Money::fromString($amount);
+
+        $result = $money->round($scale);
+
+        $this->assertEquals($expectedResult, $result->toString());
+    }
+
+    /**
+     * @dataProvider roundProvider
+     * @param string $amount
+     * @param int $scale
+     */
+    public function testRoundingIsSameAsInPhp(string $amount, int $scale): void
```

Comment:
What is the purpose of this test? I don't see it's value

Reply:
I see this as a fairly decent way of showcasing how exactly rounding works (in addition to `testRound` method itself). It asserts that the behavior hasn't changed, which might be useful eg. during business validation.

Reply:
As you said, "showing how rounding works" is already written in rounding use-case.

This test says "rounding works as same as in pure PHP", but I don't see value in such claim. I don't expect our users to use pure php rounding instead of ours. I would even suggest to use `Money` rounding exclusively, not php rounding ever.

Reply:
OK